### PR TITLE
Advisory UUID

### PR DESCRIPTION
**Pull request by bobmcwhirter**

### CODE REVIEW

**Comment by carlosthe19916** [comment]
I couldn't test this PR as when I tried to upload an advisory I got the error:
```
Query Error: error returned from database: insert or update on table "cvss3" violates foreign key constraint "cvss3_vulnerability_id_fkey"
```

I will check it again tomorrow morning

**Comment by bobmcwhirter** [comment]
Tests seem to be failing so I'll look harder in the AM.

**Comment by bobmcwhirter** [comment]
@carlosthe19916 which advisory didja ingest to result in that message?

**Comment by bobmcwhirter** [comment]
I uploaded some CSAF, OSV and CVE

image

**Comment by carlosthe19916** [comment]
> @carlosthe19916 which advisory didja ingest to result in that message?

I don't remember which one I Uploaded but this one should also generate the same error

[RHBA-2024_1440.json](https://github.com/user-attachments/files/15626729/RHBA-2024_1440.json)

**Comment by carlosthe19916** [comment]
> > @carlosthe19916 which advisory didja ingest to result in that message?
> 
> I don't remember which one I Uploaded but this one should also generate the same error
> 
> [RHBA-2024_1440.json](https://github.com/user-attachments/files/15626729/RHBA-2024_1440.json)

@bobmcwhirter my bad, the file I attached failed to be uploaded with the main branch but it worked perfectly fine with your branch/PR

**Comment by bobmcwhirter** [comment]
Works for me!

image

**Review by carlosthe19916 — APPROVED** [review]
Yes, you are right Bob. I pasted a comment just seconds before you :)

I can also confirm the uuid field was correctly added to `api/v1/advisory` and `/api/v1/advisory/{uuid}` works as expected. I'll approved the PR as it does what I was expecting from the REST point of view

File: common/src/hash.rs [hunk 1]
```diff
@@ -5,34 +5,38 @@ use std::fmt::{Display, Formatter};
 use std::str::FromStr;
 use utoipa::openapi::{Object, RefOr, Schema, SchemaType};
 use utoipa::ToSchema;
+use uuid::Uuid;
 
 #[non_exhaustive]
 #[derive(Clone, Debug, PartialEq)]
-pub enum HashKey {
+pub enum HashOrUuidKey {
```
Comment (jcrossley3) [inline common/src/hash.rs:12]: Would `Identifier` be a better name?

**Pull request merged**